src/System.Text.Encodings.Web/src/System/Text/Encodings/Web/JavaScriptStringEncoder.cs

```diff
@@ -109,14 +112,14 @@ public unsafe override bool TryEncodeUnicodeScalar(int unicodeScalar, char* buff
             // be written out as numeric entities for defense-in-depth.
             // See UnicodeEncoderBase ctor comments for more info.
 
-            if (!Encodes(unicodeScalar)) { return unicodeScalar.TryWriteScalarAsChar(buffer, bufferLength, out numberOfCharactersWritten); }
-            else if (unicodeScalar == '\b') { return b.TryCopyCharacters(buffer, bufferLength, out numberOfCharactersWritten); }
-            else if (unicodeScalar == '\t') { return t.TryCopyCharacters(buffer, bufferLength, out numberOfCharactersWritten); }
-            else if (unicodeScalar == '\n') { return n.TryCopyCharacters(buffer, bufferLength, out numberOfCharactersWritten); }
-            else if (unicodeScalar == '\f') { return f.TryCopyCharacters(buffer, bufferLength, out numberOfCharactersWritten); }
-            else if (unicodeScalar == '\r') { return r.TryCopyCharacters(buffer, bufferLength, out numberOfCharactersWritten); }
-            else if (unicodeScalar == '/') { return forward.TryCopyCharacters(buffer, bufferLength, out numberOfCharactersWritten); }
-            else if (unicodeScalar == '\\') { return back.TryCopyCharacters(buffer, bufferLength, out numberOfCharactersWritten); }
+            if (!Encodes(unicodeScalar)) { return TryWriteScalarAsChar(unicodeScalar, buffer, bufferLength, out numberOfCharactersWritten); }
+            else if (unicodeScalar == '\b') { return TryCopyCharacters(b, buffer, bufferLength, out numberOfCharactersWritten); }
+            else if (unicodeScalar == '\t') { return TryCopyCharacters(t, buffer, bufferLength, out numberOfCharactersWritten); }
+            else if (unicodeScalar == '\n') { return TryCopyCharacters(n, buffer, bufferLength, out numberOfCharactersWritten); }
+            else if (unicodeScalar == '\f') { return TryCopyCharacters(f, buffer, bufferLength, out numberOfCharactersWritten); }
+            else if (unicodeScalar == '\r') { return TryCopyCharacters(r, buffer, bufferLength, out numberOfCharactersWritten); }
+            else if (unicodeScalar == '/') { return TryCopyCharacters(forward, buffer, bufferLength, out numberOfCharactersWritten); }
+            else if (unicodeScalar == '\\') { return TryCopyCharacters(back, buffer, bufferLength, out numberOfCharactersWritten); }
```

Comment:
Seems like a fair amount of code duplication.  I wonder if it'd be better as:
```C#
if (!Encodes(unicodeScalar))
    return TryWriteScalarAsChar(unicodeScalar, buffer, buffer.Length, out numberOfCharactersWritten);

char[] toCopy = null;
switch (unicodeScalar)
{
    case '\b': toCopy = s_b; break;
    case '\t': toCopy = s_t; break;
    case '\n': toCopy = s_n; break;
    case '\f': toCopy = s_f; break;
    case '\r': toCopy = s_r; break;
    case '/': toCopy = s_forward; break;
    case '\\': toCopy = s_back; break;
    default:
        return TryWriteEncodedScalarAsNumbericEntity(unicodeScalar, buffer, bufferLength, out numberOfCharactersWritten;
}
return TryCopyCharacters(toCopy, buffer, bufferLength, out numberOfCharactersWritten);
```
or something like that.